rotkehlchen/tests/api/test_current_assets_price.py

```diff
@@ -305,3 +305,47 @@ def test_inquirer_oracles_affect_manual_price(inquirer):
     assert inquirer.find_usd_price(A_ETH) == 3  # Should remain the same since cache should be hit
     inquirer.remove_cached_current_price_entry(cache_key=(A_ETH, A_USD))
     assert inquirer.find_usd_price(A_ETH) > 100  # should have real price now
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+def test_get_all_current_prices(rotkehlchen_api_server):
+    """Test that the endpoint to fetch all manual imput returns the correct prices results"""
+    # Check that when there are no entries in the database the result is empty
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'currentassetspriceresource',
+        ),
+    )
+    result = assert_proper_response_with_result(response)
+    assert result == []
+
+    # Add prices to the database and one that overwrites another with the same from/to
+    GlobalDBHandler().add_manual_current_price(
+        from_asset=A_ETH,
+        to_asset=A_EUR,
+        price=Price(FVal(10)),
+    )
+    GlobalDBHandler().add_manual_current_price(
+        from_asset=A_ETH,
+        to_asset=A_EUR,
+        price=Price(FVal(5)),
+    )
+    GlobalDBHandler().add_manual_current_price(
+        from_asset=A_USD,
+        to_asset=A_EUR,
+        price=Price(FVal(25)),
+    )
+
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'currentassetspriceresource',
+        ),
+    )
+    result = assert_proper_response_with_result(response)
+    expected_response = [
```

Comment:
shouldn't you also check that the first ETH/EUR price becomes a historical price?

Reply:
That should be tested in a different test when adding the manual current prices. I see that is not the case so would add a test for that